inference_cli/main.py

```diff
@@ -14,6 +14,26 @@
 app.add_typer(benchmark_app, name="benchmark")
 
 
+def version_callback(value: bool):
+    if value:
+        from importlib.metadata import version, PackageNotFoundError
+
+        try:
+            inference_version = version('inference')
+            print(f"Inference Version: {inference_version}")
+        except PackageNotFoundError:
+            print(f"Inference Version: using local install")
+            
+        raise typer.Exit()
+
+@app.callback()
+def main(
+    version: Annotated[
+        Optional[bool], typer.Option("--version", callback=version_callback)
```

Comment:
Please add `--is-eager` to handle situation where other option is invoked before `--version` - as suggested by [typer docs](https://typer.tiangolo.com/tutorial/options/version/)